packages/calypso-e2e/src/lib/blocks/block-flows/all-form-fields.ts

```diff
@@ -128,10 +150,23 @@ export class AllFormFieldsFlow implements BlockFlow {
 	 *
 	 * @param {EditorContext} context The editor context object.
 	 * @param {string} blockName Name of the block.
+	 * @param {string} lastBlockName The name of the previously inserted block.
+	 * @param {boolean} isRefactor Whether the block is part of the refactored form fields.
 	 */
-	private async addFieldBlockToForm( context: EditorContext, blockName: string ) {
+	private async addFieldBlockToForm(
+		context: EditorContext,
+		blockName: string,
+		lastBlockName: string,
+		isRefactor: boolean
+	) {
 		const openInlineInserter: OpenInlineInserter = async ( editorCanvas ) => {
-			await context.editorPage.selectParentBlock( 'Form' );
+			if ( isRefactor ) {
+				await context.editorPage.selectParentBlock( lastBlockName );
+				await context.editorPage.selectParentBlock( 'Form' );
+			} else {
+				await context.editorPage.selectParentBlock( 'Form' );
+			}
```

Comment:
The refactored form fields are nested and one of the nested fields is always selected when first inserted. 

Form
  Name Field
    Label 
    Input
    
So we can't directly click the parent button for 'Form' block. We have to first  click the parent button for the Name Field block, and then the Form block.